x-pack/plugins/transform/server/routes/api/transforms.ts

```diff
@@ -0,0 +1,360 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+import { schema } from '@kbn/config-schema';
+
+import { RequestHandler } from 'kibana/server';
+import { CallCluster } from 'src/legacy/core_plugins/elasticsearch';
+import { wrapEsError } from '../../../../../legacy/server/lib/create_router/error_wrappers';
+
+import { TRANSFORM_STATE } from '../../../../../legacy/plugins/transform/public/app/common';
+import {
+  TransformEndpointRequest,
+  TransformEndpointResult,
+} from '../../../../../legacy/plugins/transform/public/app/hooks/use_api_types';
+import { TransformId } from '../../../../../legacy/plugins/transform/public/app/common/transform';
+
+import { RouteDependencies } from '../../types';
+
+import { addBasePath } from '../index';
+
+import { isRequestTimeout, fillResultsWithTimeouts, wrapError } from './error_utils';
+import { schemaTransformId, SchemaTransformId } from './schema';
+import { registerTransformsAuditMessagesRoutes } from './transforms_audit_messages';
+
+enum TRANSFORM_ACTIONS {
+  STOP = 'stop',
+  START = 'start',
+  DELETE = 'delete',
+}
+
+interface StopOptions {
+  transformId: TransformId;
+  force: boolean;
+  waitForCompletion?: boolean;
+}
+
+export function registerTransformsRoutes(routeDependencies: RouteDependencies) {
+  const { router, license } = routeDependencies;
+  router.get(
+    { path: addBasePath('transforms'), validate: {} },
```

Comment:
Nit: maybe it would be preferable to set `validate` to `false` if no validation is needed?

Reply:
Fixed in 643c4fe.